pkg/archive/paths.go

```diff
@@ -17,18 +17,48 @@ package archive
 import (
 	"archive/tar"
 	"archive/zip"
+	"compress/bzip2"
 	"compress/gzip"
 	"context"
 	"io"
 	"os"
+	"strings"
 )
 
-// A WalkFn iterates through an archive, calling FileWalkFn on each member file.
-type WalkFn func(ctx context.Context, path string, walkFn FileWalkFn) error
+const (
+	TarArchive FormatType = iota
+	TarGzArchive
+	TarBz2Archive
+	ZipArchive
+	UnsupportedArchive
```

Comment:
Probably better to have this as the first value. Advantages are:

1. Uninitialized version of type starts as "Unsupported"
2. Reduces pain of adding other types later (if `UnsupportedArchive` is first value, adding other new values is no issue, while otherwise adding new value will either shift the value of `UnsupportedArchive` to a higher one or add another one afterwards)

Reply:
Thanks, will adjust.